HW_Lect5/DGE_analysis.py

```diff
@@ -1,154 +1,177 @@
-    se = std / np.sqrt(len(sample))
-    left_b = mean - 1.96 * se
-    right_b = mean + 1.96 * se
-    return (left_b, right_b)
-
-
-def check_intervals_intersect(intervals):
-    first_ci = intervals[0]
-    second_ci = intervals[1]
-    return not ((first_ci[1] <= second_ci[0]) or (second_ci[1] <= first_ci[0]))
-
-
-def table_ci_preproc(table, celltype):
-    '''
-    Preprocesses data for CI-test (includind CI evaluation)
-    :param table:
-    :param celltype:
-    :return:
-    '''
-    table = table.iloc[:, :-1].apply(calculate_ci, axis=0).transpose()
-    table[f'{celltype}_interval'] = list(zip(table[0], table[1]))
-    table.drop([0, 1], axis=1, inplace=True)
-    return table
-
-
-def table_z_preproc(table, celltype):
-    '''
-    Preprocesses data for Z-test
-    '''
-    table = table.iloc[:, :-1].transpose()
-    table[f'{celltype}'] = table.values.tolist()
-    table.drop(list(range(0, 500)), axis=1, inplace=True)
-    return table
-
-
-def check_dge_with_ci(first_table, second_table, celltypes=("Cell_type_1", "Cell_type_2")):
-    first_table = table_ci_preproc(first_table, celltypes[0])
-    second_table = table_ci_preproc(second_table, celltypes[1])
-    intervals = pd.concat([first_table, second_table], axis=1)
-    ci_test_results = intervals.apply(check_intervals_intersect, axis=1)
-    return ci_test_results
-
-
-def check_dge_with_ztest(first_table, second_table, celltypes=("Cell_type_1", "Cell_type_2")):
-    first_table = table_z_preproc(first_table, celltypes[0])
-    second_table = table_z_preproc(second_table, celltypes[1])
-    dge_ztest = pd.concat([first_table, second_table], axis=1)
-    z_test_results = dge_ztest.apply(lambda row: ztest(row.Cell_type_1, row.Cell_type_2), axis=1)
-    dge_ztest[['z_score', 'p_value']] = pd.DataFrame(z_test_results.tolist(), index=z_test_results.index)
-    dge_ztest['Significant_differences'] = dge_ztest['p_value'] < 0.05
-    return dge_ztest['p_value'], dge_ztest['Significant_differences']
-
-
-if __name__ == '__main__':
-    output, expressions_1, expressions_2 = parsing_args()
-    to_plot = input("Do you want to get some expression plots for any gene? [y/n]: ")
-    if to_plot == 'y':
-        plotter(expressions_1, expressions_2)
-
-    ci_test_results = check_dge_with_ci(expressions_1, expressions_2)
-    z_test_p_values, z_test_results = check_dge_with_ztest(expressions_1, expressions_2)
-    mean_diff = expressions_1.mean(axis=0) - expressions_2.mean(axis=0)
-
-    results = {
-        "ci_test_results": ci_test_results,
-        "z_test_results": z_test_results,
-        "z_test_p_values": z_test_p_values,
-        "mean_diff": mean_diff
-    }
-
-    results = pd.DataFrame(results)
-    results.to_csv(f"{output}.csv")
+import numpy as np
+import pandas as pd
+import seaborn as sns
+import matplotlib.pyplot as plt
+import argparse
+from os.path import join
+from statsmodels.stats.weightstats import ztest
+from statsmodels.stats.multitest import multipletests

```

Comment:
так как ты снова все сделал так, что мне сложно придраться, то будет замечание по сортировке импортов, тут не совсем правильный порядок
```suggestion
import argparse
from os.path import join

import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
import seaborn as sns
from statsmodels.stats.multitest import multipletests
from statsmodels.stats.weightstats import ztest
```